templates/baremetal-ports-all.yaml

```diff
@@ -0,0 +1,114 @@
+heat_template_version: 2014-10-16
+
+parameters:
+
+  baremetal_prefix:
+    type: string
+
+  provision_net:
+    type: string
+
+  public_net:
+    type: string
+    description: Name of external network
+    default: public
+
+  suffix:
+    type: string
+
+  overcloud_internal_net:
+    type: string
+    description: Name of internal API network
+    default: overcloud_internal_net
+
+  overcloud_storage_net:
+    type: string
+    description: Name of storage network
+    default: overcloud_storage_net
+
+  overcloud_storage_mgmt_net:
+    type: string
+    description: Name of storage management network
+    default: overcloud_storage_mgmt_net
+
+  overcloud_tenant_net:
+    type: string
+    description: Name of tenant network
+    default: overcloud_tenant_net
+
+resources:
+
+  provision_port:
+    type: OS::Neutron::Port
+    properties:
+      name:
+        list_join:
+        - ''
+        - - {get_param: baremetal_prefix}
+          - {get_param: suffix}
+      network: {get_param: provision_net}
+
+  internal_port:
+    type: OS::Neutron::Port
+    properties:
+      name:
+        list_join:
+        - ''
+        - - internal_
+          - {get_param: baremetal_prefix}
+          - {get_param: suffix}
+      network: {get_param: overcloud_internal_net}
+
+  storage_port:
+    type: OS::Neutron::Port
+    properties:
+      name:
+        list_join:
+        - ''
+        - - storage_
+          - {get_param: baremetal_prefix}
+          - {get_param: suffix}
+      network: {get_param: overcloud_storage_net}
+
+  storage_mgmt_port:
+    type: OS::Neutron::Port
+    properties:
+      name:
+        list_join:
+        - ''
+        - - storage_mgmt_
+          - {get_param: baremetal_prefix}
+          - {get_param: suffix}
+      network: {get_param: overcloud_storage_mgmt_net}
+
+  tenant_port:
+    type: OS::Neutron::Port
+    properties:
+      name:
+        list_join:
+        - ''
+        - - tenant_
+          - {get_param: baremetal_prefix}
+          - {get_param: suffix}
+      network: {get_param: overcloud_tenant_net}
+
+  public_port:
+    type: OS::Neutron::Port
+    properties:
+      name:
+        list_join:
+        - ''
+        - - public_
+          - {get_param: baremetal_prefix}
+          - {get_param: suffix}
+      network: {get_param: public_net}
+
+outputs:
+  ports:
+    value:
+    - {port: {get_resource: provision_port}}
+    - {port: {get_resource: internal_port}}
```

Comment:
I think there's a mismatch here.  The templates appear to be using nic2 as the external network, but this is going to result in nic2 being the internal network.